pkg/controller/volume/attachdetach/attach_detach_controller_test.go

```diff
@@ -150,16 +233,90 @@ func Test_AttachDetachControllerRecovery(t *testing.T) {
 func attachDetachRecoveryTestCase(t *testing.T, extraPods1 []*v1.Pod, extraPods2 []*v1.Pod) {
 	fakeKubeClient := controllervolumetesting.CreateTestClient()
 	informerFactory := informers.NewSharedInformerFactory(fakeKubeClient, time.Second*1)
-	//informerFactory := informers.NewSharedInformerFactory(fakeKubeClient, time.Second*1)
-	plugins := controllervolumetesting.CreateTestPlugin()
-	var prober volume.DynamicPluginProber = nil // TODO (#51147) inject mock
 	nodeInformer := informerFactory.Core().V1().Nodes().Informer()
 	csiNodeInformer := informerFactory.Storage().V1().CSINodes().Informer()
 	podInformer := informerFactory.Core().V1().Pods().Informer()
-	var podsNum, extraPodsNum, nodesNum, i int
+	var extraPodsNum, nodesNum int
 
 	stopCh := make(chan struct{})
 
+	adc, plugins, _, nodesNum := testCaseSetup(t, fakeKubeClient, informerFactory, nodeInformer, csiNodeInformer, podInformer, stopCh)
+
+	// Populate ASW
+	err := adc.populateActualStateOfWorld()
```

Comment:
why was this test case modified? the diff is pretty hard to read due to the extra changes being made around here. Could you split it out into 2 commits, one for the new test and one for miscellaneous changes? In general I would avoid making miscellaneous clean-up changes when trying to add significant new logic - makes it hard for reviewers to figure out what's "new"